lib/pbench/test/functional/server/test_put.py

```diff
@@ -190,11 +191,12 @@ def test_index_all(server_client: PbenchServerClient, login_user):
                     continue
                 not_indexed.append(dataset)
                 if dataset.metadata["user.pbench.access"] == "public":
-                    assert dataset.metadata["global.pbench.origin"] == "test"
+                    assert dataset.metadata["server.origin"] == "test"
                     assert dataset.metadata["dataset.access"] == "public"
                 else:
                     assert dataset.metadata["dataset.access"] == "private"
-                    assert dataset.metadata["global.pbench.origin"] is None
+                    assert dataset.metadata["server.origin"] is None
+                    assert dataset.metadata["server.archiveonly"] is False
                     assert dataset.metadata["user.pbench.access"] is None
```

Comment:
Sure, but the datasets *alternate*; so if the user namespace really doesn't work, 196 or 197 will fail because we didn't get the matching value at 192.

Reply:
[I'm not sure which lines you're referring to...I'm thinking it's the original code rather than your updated code.]

If the `user` namespace really doesn't work (for some definition of that), then the `if` at (old) line 192 would (likely) send control to the `else` clause where the fetch of `user.pbench.access` would (likely) return `None`, and so the test wouldn't detect the problem.

(Again, this is not huge deal, but, if you're fixing other stuff, you might consider fixing this.  I _think_ that all that is required is to make the assertions in both branches of the `if` check all the same keys.)

Reply:
> I _think_ that all that is required is to make the assertions in both branches of the `if` check all the same keys.

My morning is like your afternoon, I guess.  The assertions are fine as is, except that we should be setting `user.pbench.access` to `"private"` (or similar) when it is not set to `"public"`, so that the test can assure that it has the appropriate value in both cases.  (Sorry for the wayward path, here....)

Reply:
The point was to *not* set it in one case to be sure it came back "pristine". Just can't win, I guess.

Again, I intend to write a suite of metadata test to check `PUT`/`GET` behaviors in more detail; there's no point in being completely exhaustive here, but I wanted to sprinkle in a "sampler" so it wasn't total uncovered.